### ISSUE TITLE
Check does not re-run on force push

### ISSUE THREAD

**Issue by jamesobutler** [SystangoTechnologies/commit-message-lint#19]
I've noticed that Commit Message Lint works when issuing a brand new PR in my private repo. However, if there is a force push action that occurs, the check doesn't seem to re-run and it just sits waiting.  This is a problem as I'm using this as a required-check as part of my branch rules.  Other checks are re-run following a force push.

![image](https://user-images.githubusercontent.com/15837524/92286717-cb088780-eed5-11ea-89a5-aed1bbfd34b3.png)

**Comment by sumits-systango** [comment]
We'll take a look.

**Comment by jamesobutler** [comment]
🙏 Thank you!

**Comment by sumits-systango** [comment]
@jamesobutler 
We tried reproducing this but were not able to. One possible reason you saw the issue could be with another issue we were facing with our cloud provider sometime around August end. Can you check again?

**Comment by jamesobutler** [comment]
Ok I tried some more testing again and here are my results:

Public repo, issue a PR from one branch to another within the repo, message lint: ✔️
^ then force push the branch, message lint: ✔️ 
Made repo Private, then force pushed the branch, message lint: ✔️ 

My specific case that I use:
Private repo (within organization), issue a PR from forked repo branch to main upstream branch, message lint: ✔️ 
^ then force push the forked repo's branch, message lint: ❌ (does not re-run)

When it doesn't re-run it doesn't show up in the PR checks area if message-lint is not a required check to pass before merging.  The image I provided originally showed that it was not re-running with it as a required check for merging.

**Comment by jamesobutler** [comment]
Hi @sumits-systango,

Were you able to reproduce based on the above notes?

**Comment by sumits-systango** [comment]
@jamesobutler yes, we were able to reproduce and are working on a fix. It seems to be related to fork repositories. We'll let you know as soon as a fix is deployed.

**Comment by jamesobutler** [comment]
Thanks so much! Very much appreciated.

**Comment by jamesobutler** [comment]
Hi @sumits-systango! Any update on this work? Or will this be a `wont-fix` for this project? Thanks!

**Comment by apresswala-systango** [comment]
@jamesobutler 
We are currently loaded with several requests and some upgrades. And this change is taking longer than expected. But this is still in the list and will be rolled out soon.

**Comment by renanfenrich** [comment]
any news on this?

**Comment by apresswala-systango** [comment]
Hi @renanfenrich @jamesobutler, We have deployed the change for the Issues you were having with the Forked Repositories.
Please check on your end and let us know if there are any issues.

**Comment by jamesobutler** [comment]
@apresswala-systango I force pushed a branch tied to an already open PR from a private repo fork to the upstream private repo and the commit-message-lint check was not run. I then tried opening a new PR from a private repo fork to the upstream private repo and the commit-message-lint check also did not run.

**Comment by jamesobutler** [comment]
Was this a server-side change? I see no recent pushed updates to this repo if it was for the app code.  The last commit is from May 4th 2020.

**Comment by sumits-systango** [comment]
Yes, this was a server side change @jamesobutler . We'll take a look and get back.

**Comment by jamesobutler** [comment]
@sumits-systango Any update on this?

**Comment by apresswala-systango** [comment]
Hi @jamesobutler, We are debugging this at our end. We are trying to reproduce the issue you are facing with our repositories. It would take some more time for us to get back to you on this.

**Comment by jamesobutler** [comment]
I have unfortunately uninstalled this GitHub app as it has continued to be broken.

**Comment by sumits-systango** [comment]
Hi @jamesobutler 
That's unfortunate for us. But this issue is not easily reproducible at our end, which is why it is taking so much time. I request you to try the app once again once this is fixed.

**Comment by Murtowski** [comment]
The problem I noticed is that `commit message lint app` caches previous commit messages, never clear it, which leads to bugs.
Imagine I commit message wrongly first time and I then rename with using `rebase` or `amend` but  app doesn't refresh previous commit messages list and answer with error - which is wrong.
